pcsx2/DEV9/ATA/ATA_Info.cpp

```diff
@@ -0,0 +1,391 @@
+	 */
+	int index = 0;
+	WriteUInt16(identifyData, &index, 0x0040); //word 0
+	//Default Num of cylinders
+	WriteUInt16(identifyData, &index, defCylinders); //word 1
+	//Specific configuration
+	index += 1 * 2; //word 2
+	//Default Num of heads (Retired)
+	WriteUInt16(identifyData, &index, defHeads); //word 3
+	//Number of unformatted bytes per track (Retired)
+	WriteUInt16(identifyData, &index, (u16)(sectorSize * defSectors)); //word 4
+	//Number of unformatted bytes per sector (Retired)
+	WriteUInt16(identifyData, &index, sectorSize); //word 5
+	//Default Number of sectors per track (Retired)
+	WriteUInt16(identifyData, &index, defSectors); //word 6
+	//Reserved for assignment by the CompactFlash� Association
+	index += 2 * 2; //word 7-8
+	//Retired
+	index += 1 * 2; //word 9
+	//M-Serial number (20 ASCII characters)
+	WritePaddedString(identifyData, &index, "PCSX2-DEV9-ATA-HDD", 20); //word 10-19
+	//Buffer(cache) type (Retired)
+	WriteUInt16(identifyData, &index, /*3*/ 0); //word 20
+	//Buffer(cache) size in sectors (Retired)
+	WriteUInt16(identifyData, &index, /*512*/ 0); //word 21
+	//Number of ECC bytes available on read / write long commands (Obsolete)
+	WriteUInt16(identifyData, &index, /*4*/ 0); //word 22
+	//M-Firmware revision (8 ASCII characters)
+	WritePaddedString(identifyData, &index, "FIRM100", 8); //word 23-26
+	//M-Model number (40 ASCII characters)
+	WritePaddedString(identifyData, &index, "PCSX2-DEV9-ATA-HDD", 40); //word 27-46
+	//M-READ/WRITE MULI max sectors
+	WriteUInt16(identifyData, &index, 128 & (0x80 << 8)); //word 47
+	//Dword IO supported
+	WriteUInt16(identifyData, &index, 1); //word 48
+	//M-Capabilities
+	/*
+	 * bits 7-0: Retired
+	 * bit 8: DMA supported
+	 * bit 9: LBA supported
+	 * bit 10:IORDY may be disabled
+	 * bit 11:IORDY supported
+	 * bit 12:Reserved
+	 * bit 13:Standby timer values as specified in this standard are supported
+	 */
+	WriteUInt16(identifyData, &index, ((1 << 11) | (1 << 9) | (1 << 8))); //word 49
+	//M-Capabilities (0-Shall be set to one to indicate a device specific Standby timer value minimum)
+	index += 1 * 2; //word 50
+	//PIO data transfer cycle timing mode (Obsolete)
+	WriteUInt16(identifyData, &index, (u8)((pioMode > 2 ? pioMode : 2) << 8)); //word 51
+	//DMA data transfer cycle timing mode (Obsolete)
+	WriteUInt16(identifyData, &index, 0); //word 52
+	//M
+	/*
+	 * bit 0: Fields in 54:58 are valid (Obsolete)
+	 * bit 1: Fields in 70:64 are valid
+	 * bit 2: Fields in 88 are valid
+	 */
+	WriteUInt16(identifyData, &index, (1 | (1 << 1) | (1 << 2))); //word 53
+	//Number of current cylinders
+	WriteUInt16(identifyData, &index, curCylinders); //word 54
+	//Number of current heads
+	WriteUInt16(identifyData, &index, curHeads); //word 55
+	//Number of current sectors per track
+	WriteUInt16(identifyData, &index, curSectors); //word 56
+	//Current capacity in sectors
+	WriteUInt32(identifyData, &index, (u32)curOldsize); //word 57-58
+	//M
```

Comment:
typo? happens multiple times

Reply:
The `//M` comment? It's to to with ATA Flash memory (M-Flash in the standards iirc), which I'm not supporting here.
I would need to refresh myself on what the full name of the field is.

Reply:
Nope, the M was something I did to keep track of something, but forget what is was I put it in for, will be amending that